run_robot_test.sh

```diff
@@ -107,3 +107,35 @@ TEST_ARTIFACT_DIR=$(mktemp -d -p ${TEST_ARTIFACT_DIR} -t ods-ci-$(date +%Y-%m-%d
 ./venv/bin/robot -d ${TEST_ARTIFACT_DIR} -x xunit_test_result.xml -r test_report.html ${TEST_VARIABLES} --variablefile ${TEST_VARIABLES_FILE} --exclude TBC ${EXTRA_ROBOT_ARGS} ${TEST_CASE_FILE}
 
 esac
+
+case "$(uname -s)" in
```

Comment:
I think this 7 lines (111 to 117) is the only change required and the rest is duplicated code. Can you add just this?

Reply:
Please find below constraint while performing the above task in same case block:
1.  line no 68 -> mac os doesn't support "lsb_release" command. (lsb_release is a command that runs on Linux only: the "LSB" stands for "Linux Standard Base" and this refers to a standard baseline set of tools and libraries that Linux distributions can elect to support )
2. line no 104 -> mktemp command in Mac OS doesn't support -p parameter. Checked manual for mktemp for mac.

Reply:
What I was proposing you was to reduce the replicated code when possible. Something like this:

```
case "$(uname -s)" in
    Darwin)
         echo "MACOS"
         echo "setting driver  to $currentpath/Drivers/MACOS"
         PATH=$PATH:$currentpath/drivers/MACOS
         export PATH=$PATH
         echo "$PATH"
         ;;
    Linux)
       case "$(lsb_release --id --short)" in
       "Fedora"|"CentOS")
             ## Bootstrap script to setup drivers ##
             echo "setting driver  to $currentpath/Drivers/fedora"
             PATH=$PATH:$currentpath/drivers/fedora
             export PATH=$PATH
             echo $PATH
        ;;
        "Ubuntu")
             echo "Not yet supported, but shouldn't be hard for you to fix :) "
             echo "Please add the driver, test and submit PR"
             exit 1
        ;;
        "openSUSE project"|"SUSE LINUX"|"openSUSE")
             echo "Not yet supported, but shouldn't be hard for you to fix :) "
             echo "Please add the driver, test and submit PR"
             exit 1
        ;;
        esac
esac
```

And then 

```
#TODO: Configure the "tmp_dir" creation so that we can have a "latest" link
# mktemp command in Mac OS doesn't support -p parameter
case "$(uname -s)" in
    Darwin)
        TEST_ARTIFACT_DIR=$(mktemp -d  ${TEST_ARTIFACT_DIR} -t ods-ci-$(date +%Y-%m-%d-%H-%M)-XXXXXXXXXX)
         ;;
    Linux)
        TEST_ARTIFACT_DIR=$(mktemp -d -p ${TEST_ARTIFACT_DIR} -t ods-ci-$(date +%Y-%m-%d-%H-%M)-XXXXXXXXXX)
        ;;
esac

```